src/plugins/discover/public/application/view_components/utils/use_dataset_manager.ts

```diff
@@ -5,35 +5,35 @@
 
 import { useState, useEffect } from 'react';
 import { Subscription } from 'rxjs';
-import { DataSetManager } from '../../../../../data/public';
 import { Dataset } from '../../../../../data/common';
+import { DatasetContract } from '../../../../../data/public';
 
-interface UseDataSetManagerProps {
-  dataSet?: Dataset;
-  dataSetManager: DataSetManager;
+interface UseDatasetManagerProps {
+  dataset?: Dataset;
+  datasetManager: DatasetContract;
 }
 
-export const useDataSetManager = (props: UseDataSetManagerProps) => {
-  const [dataSet, setDataSet] = useState<Dataset | undefined>(
-    props.dataSet || props.dataSetManager.getDataSet()
+export const useDatasetManager = (props: UseDatasetManagerProps) => {
```

Comment:
If we want to keep this instead of using the one from the data plugin, lets rename this hook too
```suggestion
export const useDataset = (props: UseDatasetManagerProps) => {
```